src/main/java/org/folio/rest/impl/HoldingsStorageAPI.java

```diff
@@ -408,4 +414,31 @@ private boolean isUUID(String id) {
       return false;
     }
   }
+
+  private void updateItemEffectiveCallNumbersByHoldings(HoldingsRecord holdingsRecord, Map<String, String> okapiHeaders, Context vertexContext) {
+    String query = String.format("holdingsRecordId==%s", holdingsRecord.getId());
+    log.info(query);
+    PgUtil.get(ITEM_TABLE, Item.class, Items.class, query, 0, -1, okapiHeaders, vertexContext, GetItemStorageItemsResponse.class, response -> {
+      if (response.succeeded()) {
+        updateEffectiveCallNumbers((Items) response.result().getEntity(), holdingsRecord, okapiHeaders, vertexContext);
+      }
+    });
+  }
+
+  private void updateEffectiveCallNumbers(Items items, HoldingsRecord holdingsRecord, Map<String, String> okapiHeaders,
+      Context vertexContext) {
+    items.getItems().forEach(i -> {
+      String updatedCallNumner = null;
+      if (i.getItemLevelCallNumber() != null && !i.getItemLevelCallNumber().isEmpty()) {
+          updatedCallNumner = i.getItemLevelCallNumber();
+      } else if (i.getHoldingsRecordId() != null && !i.getHoldingsRecordId().isEmpty()) {
```

Comment:
Is it fine that we are checking that `i.getHoldingsRecordId() != null` but accessing to the `holdingsRecord`

Reply:
This should have been checking the holding record's call number. I've updated the branch to correct this.